File: dataset/CoT/GEMINI_gemini-2.5-pro/Task16.py
def solution(string, ending):
    """
    Checks if the first string ends with the second string.

    :param string: The main string.
    :param ending: The string to check for at the end of the main string.
    :return: True if the string ends with the ending, False otherwise.
    """
    if not isinstance(string, str) or not isinstance(ending, str):
        return False
    return string.endswith(ending)

if __name__ == '__main__':
    # Test cases
    test_cases = [
        ("abc", "bc", True),
        ("abc", "d", False),
        ("samurai", "ai", True),
        ("abc", "", True),
        ("abc", "abcde", False)
    ]

    for i, (str_val, end_val, expected) in enumerate(test_cases):
        print(f"Test Case {i + 1}: solution(\"{str_val}\", \"{end_val}\")")
        print(f"Expected: {expected}")
        actual = solution(str_val, end_val)
        print(f"Actual: {actual}")
        print()

File: dataset/CoT/GEMINI_gemini-2.5-pro/Task16.java
public class Task16 {

    /**
     * Checks if the first string ends with the second string.
     *
     * @param str    The main string.
     * @param ending The string to check for at the end of str.
     * @return true if str ends with ending, false otherwise.
     */
    public static boolean solution(String str, String ending) {
        if (str == null || ending == null) {
            return false;
        }
        return str.endsWith(ending);
    }

    public static void main(String[] args) {
        // Test cases
        System.out.println("Test Case 1: solution(\"abc\", \"bc\")");
        System.out.println("Expected: true");
        System.out.println("Actual: " + solution("abc", "bc"));
        System.out.println();

        System.out.println("Test Case 2: solution(\"abc\", \"d\")");
        System.out.println("Expected: false");
        System.out.println("Actual: " + solution("abc", "d"));
        System.out.println();

        System.out.println("Test Case 3: solution(\"samurai\", \"ai\")");
        System.out.println("Expected: true");
        System.out.println("Actual: " + solution("samurai", "ai"));
        System.out.println();

        System.out.println("Test Case 4: solution(\"abc\", \"\")");
        System.out.println("Expected: true");
        System.out.println("Actual: " + solution("abc", ""));
        System.out.println();

        System.out.println("Test Case 5: solution(\"abc\", \"abcde\")");
        System.out.println("Expected: false");
        System.out.println("Actual: " + solution("abc", "abcde"));
        System.out.println();
    }
}